substrate/frame/support/src/storage/types/counted_map.rs

```diff
@@ -47,6 +47,33 @@ use sp_std::prelude::*;
 ///
 /// Whenever the counter needs to be updated, an additional read and write occurs to update that
 /// counter.
+///
+/// For general information regarding the `#[pallet::storage]` attribute, refer to 
+/// [`crate::pallet_macros::storage`].
+///
+/// # Example
+///
+/// ```
+/// #[frame_support::pallet]
+/// mod pallet {
+///     # use frame_support::pallet_prelude::*;
+///     # #[pallet::config]
+///     # pub trait Config: frame_system::Config {}
+///     # #[pallet::pallet]
+///     # pub struct Pallet<T>(_);
+/// 	/// A kitchen-sink counted storage map, with all possible additional attributes.
+///     #[pallet::storage]
+/// 	#[pallet::getter(fn foo)]
+/// 	#[pallet::storage_prefix = "OtherFoo"]
+/// 	#[pallet::unbounded]
+///     pub type Foo<T> = CountedStorageMap<
+/// 		Hasher = Blake2_128Concat,
+/// 		Key = u32,
+/// 		Value = u32,
+/// 		QueryKind = ValueQuery
+/// 	>;
```

Comment:
nice, I would experiment with now writing a simple (and mostly hidden) `fn main` that spins up a `TestExternalities`, add a couple of items to this map, then call `count`. 

This might exist in some of the tests in the bottom of this file, ergo you could also use `docify`.